### PR TITLE
itest+rpcserver: fix itest flake in `3rd_party_anchor_spend`

### PR DESCRIPTION
**Pull request by yyforyongyu**
Found a new flake from [this build](https://github.com/lightningnetwork/lnd/actions/runs/4422237723/jobs/7753851726):

```
harness_rpc.go:97: 
        	Error Trace:	/home/runner/work/lnd/lnd/itest/harness_rpc.go:97
        	            				/home/runner/work/lnd/lnd/itest/lnd.go:372
        	            				/home/runner/work/lnd/lnd/itest/lnd_onchain_test.go:443
        	            				/home/runner/work/lnd/lnd/itest/harness.go:286
        	            				/home/runner/work/lnd/lnd/itest/lnd_test.go:136
        	Error:      	Received unexpected error:
        	            	rpc error: code = Unknown desc = reserved wallet balance invalidated: transaction would leave insufficient funds for fee bumping anchor channel closings (see debug log for details)
        	Messages:   	Alice: failed to call SendCoins
    harness.go:342: finished test: 3rd_party_anchor_spend, start height=506, end height=514, mined blocks=8
    harness.go:348: test failed, skipped cleanup
```

This PR fixes it and adds more debug logs to the rpcserver.

### CODE REVIEW

File: docs/release-notes/release-notes-0.16.1.md [hunk 1]
```diff
@@ -5,3 +5,9 @@
 - The `lncli wallet psbt fund` command now allows users to specify the
   [`--min_confs` flag](https://github.com/lightningnetwork/lnd/pull/7510).
 
+
+## RPC
+
+- A [debug log](https://github.com/lightningnetwork/lnd/pull/7514) has been
```
Comment (positiveblue) [inline docs/release-notes/release-notes-0.16.1.md:11]: Really nice, this is something that we can add to other RPC servers :+1:

**Review by positiveblue — APPROVED** [review]

**Review by guggero — COMMENTED** [review]
Great find on the itest!
More logging is definitely a good idea, what do you think of making it more generic though?

File: rpcserver.go [hunk 2]
```diff
@@ -1131,6 +1131,8 @@ func (r *rpcServer) sendCoinsOnChain(paymentMap map[string]int64,
 func (r *rpcServer) ListUnspent(ctx context.Context,
 	in *lnrpc.ListUnspentRequest) (*lnrpc.ListUnspentResponse, error) {
 
+	rpcsLog.Debug("[listunspent] request")
```
[the remark this answers is not in the document]
Comment (yyforyongyu) [reply rpcserver.go]: added it at two places, one for unary and one for stream. Tried them locally and I like the new logs, thanks for the advice!

File: rpcperms/interceptor.go [hunk 3]
```diff
@@ -772,6 +772,8 @@ func (r *InterceptorChain) rpcStateUnaryServerInterceptor() grpc.UnaryServerInte
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo,
 		handler grpc.UnaryHandler) (interface{}, error) {
 
+		r.rpcsLog.Debugf("[%v] requested", info.FullMethod)
```
Comment (guggero) [inline rpcperms/interceptor.go:775]: Nice! Not sure if this maybe gets too noisy on nodes with Lit, LNDg and other tool running. So was thinking maybe we could create a new logger just for this? Or maybe it's enough to just set `RPCS=info` if things become too noisy, looks like we don't use the `rpcsLog` for debug or trace outputs in that many places.

**Review by guggero — APPROVED** [review]
LGTM :tada:

File: rpcperms/interceptor.go (hunk 3 above)
Comment (yyforyongyu) [reply rpcperms/interceptor.go:775]: yeah we can always add a new one when it's too noisy🎧

**Pull request merged**